Sources/Build/BuildPlan/BuildPlan.swift

```diff
@@ -692,6 +727,203 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
+                    module: plugin,
+                    action: .createBuildToolCommands(
+                        package: package,
+                        target: module,
+                        pluginGeneratedSources: pluginDerivedSources.paths,
+                        pluginGeneratedResources: pluginDerivedResources.map(\.path)
+                    ),
+                    buildEnvironment: buildParameters.buildEnvironment,
+                    scriptRunner: configuration.scriptRunner,
+                    workingDirectory: package.path,
+                    outputDirectory: pluginOutputDir,
+                    toolSearchDirectories: [buildParameters.toolchain.swiftCompilerPath.parentDirectory],
+                    accessibleTools: accessibleTools,
+                    writableDirectories: writableDirectories,
+                    readOnlyDirectories: readOnlyDirectories,
+                    allowNetworkConnections: [],
+                    pkgConfigDirectories: pkgConfigDirectories,
+                    sdkRootPath: buildParameters.toolchain.sdkRootPath,
+                    fileSystem: fileSystem,
+                    modulesGraph: modulesGraph,
+                    observabilityScope: observabilityScope,
+                    completion: $0
+                )
+            }
+
+            if surfaceDiagnostics {
+                let diagnosticsEmitter = observabilityScope.makeDiagnosticsEmitter {
+                    var metadata = ObservabilityMetadata()
+                    metadata.moduleName = module.name
+                    metadata.pluginName = result.plugin.name
+                    return metadata
+                }
+
+                for line in result.textOutput.split(whereSeparator: { $0.isNewline }) {
+                    diagnosticsEmitter.emit(info: line)
+                }
+
+                for diag in result.diagnostics {
+                    diagnosticsEmitter.emit(diag)
+                }
+            }
+
+            // Add a BuildToolPluginInvocationResult to the mapping.
+            buildToolPluginResults.append(result)
+        }
+
+        return buildToolPluginResults
+    }
+
+    /// Runs any prebuild commands associated with the given list of plugin invocation results,
+    /// in order, and returns the results of running those prebuild commands.
+    fileprivate static func runPrebuildCommands(
+        using pluginConfiguration: PluginConfiguration,
+        for pluginResults: [BuildToolPluginInvocationResult],
+        fileSystem: any FileSystem,
+        observabilityScope: ObservabilityScope
+    ) throws -> [PrebuildCommandResult] {
+        // Run through all the commands from all the plugin usages in the target.
+        try pluginResults.map { pluginResult in
+            // As we go we will collect a list of prebuild output directories whose contents should be input to the
+            // build,
+            // and a list of the files in those directories after running the commands.
```

Comment:
comment formatting nit
```suggestion
            // As we go we will collect a list of prebuild output directories whose contents should be input to the
            // build, and a list of the files in those directories after running the commands.
```

Reply:
Interestingly this is what `swiftformat` did...

Reply:
Yes, its comments line length reflow needs a second pass by a human 😅

Reply:
✅